### PR TITLE
Listen to clippy

### PR DESCRIPTION
**Pull request by marceline-cramer**
Follow-up to #108. Cleans a few compiler and clippy warnings.

### CODE REVIEW

File: crates/hearth-cognito/src/lib.rs [hunk 1]
```diff
@@ -525,6 +525,21 @@ pub struct WasmPlugin {
     engine: Arc<Engine>,
 }
 
+impl Default for WasmPlugin {
+    fn default() -> Self {
+        let mut config = Config::new();
```
Comment (airidaceae) [inline crates/hearth-cognito/src/lib.rs:530]: Is there a reason we aren't using wasmtime config defaults with `Config::default()` here?

Comment (airidaceae) [reply crates/hearth-cognito/src/lib.rs:530]: I guess we are changing things so we aren't using the defaults but starting with the defaults and changing them may be better practice. It also doesn't seem to affect anything other than the strategy variable right now so its probably not a big deal.

**Review by airidaceae — APPROVED** [review]
Look over my comments and see if you agree or disagree. Feel free to pull it in if you disagree.

File: crates/hearth-cognito/src/lib.rs (hunk 1 above)
Comment (marceline-cramer) [reply crates/hearth-cognito/src/lib.rs:530]: `Config::default()` actually just calls `Config::new()`, so they're semantically equivalent.

**Pull request merged**